rustfmt-core/rustfmt-lib/src/expr.rs

```diff
@@ -1990,6 +1990,18 @@ fn choose_rhs<R: Rewrite>(
             let before_space_str = if has_rhs_comment { "" } else { " " };
 
             match (orig_rhs, new_rhs) {
+                (Some(ref orig_rhs), Some(_))
+                    if context.config.wrap_long_multiline_chains()
+                        && orig_rhs.matches('.').collect::<Vec<_>>().len() > 1 =>
```

Comment:
This check is not going to be exclusive to identifying chains and could spuriously be applied on non-chains. For a contrived example:
```rust
    let x = foobar_longgggggggggggggggggggggggggggggggggggggggggggggg(|bar| {
    // foo.
            // bar.
        // baz.
        ....
    });
```

would get formatted to the below snippet when the proposed `wrap_long_multiline_chains` is `true`, and this would be bad formatting that fails to comply with the style guide requirements.

```rust
    let x =
        foobar_longgggggggggggggggggggggggggggggggggggggggggggggg(|bar| {
        // foo.
        // bar.
        // baz.
        ....
    });
```

This is also a relatively hot path in the code, and we should avoid having construct specific (chains) logic here